packages/manager/apps/web/client/app/email-domain/general-informations/email-domain-general-informations.controller.js

```diff
@@ -125,5 +121,9 @@ angular.module('App').controller(
         this.loading.urls = false;
       });
     }
+
+    goToUpgrade() {
+      this.$state.go('app.email.domain.upgrade', { productId: this.$stateParams.productId });
```

Comment:
Could this be injected through the resolve and then the controller's constructor ?

Reply:
Its the legacy code, I need to make lot of changes to do that. Should I target this as part of this?

Reply:
If this feature is not urgent then maybe you could upgrade this page while you are at it, if it's not too much work though 🙂

Reply:
it's urgent, its already delayed

Reply:
Then let's keep that suggestion in mind for refactoring in the near future

Reply:
sure